src/Functions/generateSnowflakeID.cpp

```diff
@@ -165,24 +165,36 @@ class FunctionGenerateSnowflakeID : public IFunction
     {
         FunctionArgumentDescriptors mandatory_args;
         FunctionArgumentDescriptors optional_args{
-            {"expr", nullptr, nullptr, "Arbitrary expression"}
+            {"expr", nullptr, nullptr, "Arbitrary expression"},
+            {"machine_id", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isNativeUInt), nullptr, "UInt*"}
         };
         validateFunctionArguments(*this, arguments, mandatory_args, optional_args);
 
         return std::make_shared<DataTypeUInt64>();
     }
 
-    ColumnPtr executeImpl(const ColumnsWithTypeAndName & /*arguments*/, const DataTypePtr &, size_t input_rows_count) const override
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
         auto col_res = ColumnVector<UInt64>::create();
         typename ColumnVector<UInt64>::Container & vec_to = col_res->getData();
+        vec_to.resize(input_rows_count);
```

Comment:
Please remove l. 182-192 and add this before l. 196:

```cpp
uint64_t machine_id = getMachineId();
if (arguments.size() == 2)
{
    machine_id = arguments[1].column->getUInt(0);
    machine_id &= (1ull << machine_id_bits_count) - 1;
}
```

Reply:
Hi! For this one, I make changes based on your suggestions, but I think we have to add ``` && input_rows_count > 0``` in the if statement as well?